flink-core/src/test/java/org/apache/flink/api/connector/source/lib/NumberSequenceSourceTest.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.connector.source.lib;
+
+import org.apache.flink.api.connector.source.SourceReader;
+import org.apache.flink.api.connector.source.mocks.TestingReaderContext;
+import org.apache.flink.api.connector.source.mocks.TestingReaderOutput;
+import org.apache.flink.core.io.InputStatus;
+
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.fail;
+
+/**
+ * Tests for the {@link NumberSequenceSource}.
+ */
+public class NumberSequenceSourceTest {
+
+	@Test
+	public void testReaderCheckpoints() throws Exception {
```

Comment:
Should we also have an IT case for the source as well? This would help reveal issues earlier if this source is intended to be used by other test cases.

Reply:
I was contemplating this and decided against that, as mentioned in the PR description.

The Integration Test would mostly test the runtime source operator and coordinator, not specific to this source here.
Having at least one case of the source being used end to end makes total sense, and this is the case in other tests (see for example #13513)

Reply:
Adding an ITCase is certainly possible, it would not be much work, it would just be added testing time for something that is already covered.

Reply:
It seems that we should not reduce test time by skipping reasonable tests? The concern I has was that if there is another test using the `NumberSequenceSource` failed due to a bug in this source. The author of that test may have to look into the code of this source, which seems something we would like to avoid by having tests here.

Reply:
Fair enough. I'll add an ITCase for the NumberSequenceSource to cover the iterator sources.
Further iterator sources should then by fine with unit tests only.